### PR TITLE
test api token and refresh token

### PR DESCRIPTION
**Pull request by TheMattSchiller**
ensures that an API token as well as a refresh token can be used

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -104,6 +104,10 @@ func getAccessTokenFromJSON(body []byte) (accessToken string, err error) {
 		err = errors.WithMessage(err, "unable to unmarshall the accessToken, "+string(body))
 		return
 	}
+	if accessTokenStruct.Message == "Unauthorized access" {
```
Comment (yoshiotu) [inline client/client.go:107]: I think the message returned by the server has "Unauthorized access" as a substring, but the full message is longer.

Would it be better to check the http response status code? It looks like the status code is checked in the function `exhangeRefreshTokenForAccessToken()` on line 84 above, but there's a bug. Line 85 sets the return param `err` but then the next line is missing `return`.

One other nit: line 85 is using `errors.New(fmt.Sprintf(...))` which can be simplified to `errors.Errorf(...)`

Comment (yoshiotu) [reply client/client.go:107]: Correction: the message is "Unauthorized access" if the "refresh token" includes < 2 "." (period) characters. Seems fragile to depend on that.

Comment (TSFenwick) [reply client/client.go:107]: +1

Comment (TSFenwick) [reply client/client.go:107]: @TheMattSchiller @yoshiotu i'd be more than happy to chat with you two about this project

**Pull request closed without merging**